### PR TITLE
Utility Functions - Copy to clipboard

### PR DESCRIPTION
**Pull request by thomaslennan**
## Description

Helper function to copy text to system clipboard.

Fixes https://github.com/Teradata/covalent/issues/1438

### What's included?
- Reverted scaffolding for separate utility package
- Added 'functions' subdirectory under core/common
- Added and exported copyToClipboard function
- Added sample component that initially is used to document how to use copyToClipboard
function but subsequently can be utilized for other future functions.

#### Test Steps
- [ ] `npm run serve`
- [ ] Confirmed sample component content looks correct and copies to clipboard as expected
- [ ] Confirmed built core package included copyToClipboard function
- [ ] Confirmed function could be imported via '@covalent/core/common'

#### General Tests for Every PR

- [ ] `npm run serve:prod` still works.
- [ ] `npm run tslint` passes.
- [ ] `npm run stylelint` passes.
- [ ] `npm test` passes and code coverage is not lower.
- [ ] `npm run build:lib` still works.

##### Screenshots or link to StackBlitz/Plunker

Issue1438

### CODE REVIEW

File: src/platform/core/common/functions/clipboard.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  // Create a temporary textarea element and append to DOM
+  const fakeTextArea: HTMLTextAreaElement = document.createElement('textarea');
+  document.body.appendChild(fakeTextArea);
+
+  // Assign value to be copied to clipboard
+  fakeTextArea.value = value;
+  fakeTextArea.select();
+
+  // Copy to clipboard
+  document.execCommand('copy');
```
Comment (christianmemije) [inline src/platform/core/common/functions/clipboard.ts]: Can we return the value of this which is a boolean whether the action was successful?

Comment (thomaslennan) [reply src/platform/core/common/functions/clipboard.ts]: Done

File: src/app/components/components/functions/functions.component.html [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                <textarea matInput
+                  [(ngModel)]="copytext"
+                  placeholder="Text to be copied"
+                  rows="1"
+                  name="copytextinput">
+                </textarea>
+              </mat-form-field>
+            </div>
+          </form>
+          <button mat-button
```
Comment (christianmemije) [inline src/app/components/components/functions/functions.component.html]: Small suggestion would be to move this within the `mat-card-actions` section.

Comment (thomaslennan) [reply src/app/components/components/functions/functions.component.html]: Done

**Pull request merged**